### DISCUSSION TITLE
Project not accessible on specific network providers in Nigeria

### DISCUSSION THREAD

**Discussion by lordelogos** [vercel/community#3630] [Help] [answered]
### Summary

I am reaching out with regards to an issue I am facing with my deployment on vercel for [https://clientmail.xyz](https://clientmail.xyz)

I regularly face this issue of it not being accessible when i'm connected to my home wifi but it works fine once I switch to my personal hotspot (for context, wifi - MTN NG, hotspot- Airtel NG)  

I used to believe it was a general issue as it affected my hashnode blog before and they reached out and said it was from vercel but now I see Dub (by 
[@steventey](https://twitter.com/steventey)
) still works even over my wifi connection and the other deployment specific links work fine just not by custom domain. 

I know it will be hard to replicate this issue so I am available to offer help in anyway to ensure this issue gets fixed

### Steps to Reproduce

It is hard to replicate as it only happens sometimes over the MTN network in Nigeria.

**Comment by WorkMender** [comment]
I am facing the same issue here. However, were you able to get this working?

**Comment by rowend36** [comment]
The reason is because Vercel's shared IP address 76.76.21.21 is being blocked every now and then by ISPs. As of today 15/03/24 3:58pm, only Globacom is allowing it.

For now, there are two solutions:
 1. Move your entire site away from Vercel.
 2. Create a "redirect rule" or it's equivalent using something like Cloudflare. The purpose of the script is to redirect domain.com to www.domain.com. This works for me.

**Reply by WorkMender** [reply]
We have done so by moving to AWS amplify. There are some tradeoffs but better than nothing

**Reply by uzoway** [reply]
Hello @WorkMender & @rowend36, I'd appreciate it if you can share a resource that shows how to "create a redirect rule"

**Reply by WorkMender** [reply]
Do you intend to use AWS amplify?

**Reply by WorkMender** [reply]
You can refer to this youtube [link](https://www.youtube.com/watch?v=0B-hMvMggm8&t=307s&pp=ygULYXdzIGFtcGxpZnk%3D) to learn how to setup AWS amplify

**Reply by uzoway** [reply]
No, I don't intend to use AWS. Thank you sharing the Youtube link. Appreciate it!

**Reply by boyepanthera** [reply]
@rowend36 I did the redirect rule but it only seemed to work on desktop it doesn't redirect to www. on mobile

**Comment by rowend36** [comment]
Sure, Mr. Uzorchukwu.

But first of all, you have to have your domain hosted on Cloudflare. You'll
have to change your authoritative nameservers.
1. Sign up on Cloudflare
2. Add Website
3. Enter your domain name. You will be given 2 nameservers.
4. Find out how to change authoritative nameservers on your own for the
platform on the platform where you bought your domain.
5. Once your website has been connected to Cloudflare, follow the steps
here to create a
"dynamic" redirect rule:
https://developers.cloudflare.com/rules/url-forwarding/single-redirects/examples/
.




NB: There are in fact, other ways to do this outside Cloudflare but to keep
it simple, let's just go with Cloudflare.
Owologba Oro

Software Developer, Reelest Studios

**Reply by uzoway** [reply]
Thank you for sharing! I appreciate it.

I got the domain for the website I'm having this issue with GoDaddy. I'll have to find a way to do this on GoDaddy!
Thank you once again.

**Reply by rowend36** [reply]
It does not actually matter where you got the domain.

Just change your authoritative nameservers to the one Cloudflare gives you.

**Reply by ihtishamKhan** [reply]
@rowend36 I have changed the nameservers in goddady to Cloudflare given, but my website is hosted in vercel. How do I direct there?

**Comment by nwazuo** [comment]
I'm experiencing this issue as well. My client's websites hosted on Vercel all have frequent downtimes from the most common network providers around here. Accessing with a VPN does work but most people don't use that. I'll need to use a different service like AWS amplify going forward as this might be hard for Vercel to fix.

**Reply by uzomao** [reply]
Experiencing the same issue here, mostly with MTN NG. Would a permanent fix be the responsibility of Vercel or the ISPs?

**Comment by rowend36** [comment]
Please, follow the link on step 5.

**Comment by nwazuo** [comment]
So, I've been able to work around this using the flattened CNAME records feature of Cloudflare. 

TLDR(Too Long, Didnt' Read...): You can use Cloudflare DNS as your authoritative DNS server, move all your DNS records over to Cloudflare and configure your apex domain to use CNAME instead of A record, which helps bypass this issue. 

Basically, you can only point an A record to a specific IP address, which most of us do when we configure the domain for our Vercel websites. (i.e. pointing yyyy.com -> 76.76.21.21 on your domain registrar (e.g. GoDaddy, Namecheap, assuming you're using them as your authoritative NameServers)). 

The www version of your Vercel website domain configuration uses a CNAME record that points to cname.vercel-dns.com. Now, this CNAME record resolves to a dynamic IP address 76.76.21.xxx, and hardly every hits the problematic one 76.76.21.21. 

The trick was to find a way to get the apex domain (yourwebsite.com, without www) to resolve to one of Vercel's non-problematic IPs. 
Luckily, Cloudflare implements CNAME flattening, which allows you point your apex/root domain to a non-IP value. 

So, I point my apex domain (yourwebsite.com) to a CNAME with the `cname.vercel-dns.com` value and Cloudflare resolves that to one of Vercel's non-problematic IP addresses. 

If you're not well-versed with how DNS works, you can take some time to read(or... ask AI🙂). But essentially, you can use Cloudflare DNS as your authoritative DNS server and move all your DNS records over to Cloudflare and configure your apex domain to use CNAME instead of A record, which helps bypass this issue and keeps everything stable (also double check your other service to ensure they work, but they should).

**Reply by nwazuo** [reply]
Also, if you do this, turn off proxying on the CNAME records pointing to your apex domain and www(for your apex domain) as I experienced an 'ERR_TOO_MANY_REDIRECTS' issue with proxying on for those. 

Hope this helps.

**Reply by mosessmax** [reply]
if you get "ERR_TOO_MANY_REDIRECTS" issue on anywebsite hosted with cloudflare, just go to the SSL Status on the sidebar and set the SSL from flexible to full. and you're good to go. @nwazuo

**Reply by nwazuo** [reply]
@mosessmax It was already set to that and still not working. Turning off proxying was what worked for me

**Comment by amyegan** [comment, marked as answer]
Thank you all for for reaching out and sharing what worked for you. 

We are in contact with the Nigerian government and ISPs to remove the block block against our 76.76.21.21 IP address. Given this is not a block initiated by Vercel, we are limited in what actions we can take and I am not able to estimate long it will take to resolve.

I will post updates here as new information is received. I appreciate all of you for understanding.

**Comment by Ayodimeeji** [comment]
Add CNAME record name www.domain.xyz to your authoritative server.
use any DNS resolver to resolve cname.vercel-dns.com
Change your primary domain value to any IP address closer to your location.
Redirect domain.xyz to www.domain.xyz on vercel domain configs (Use permanent redirection)
